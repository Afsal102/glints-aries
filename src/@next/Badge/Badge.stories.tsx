import { Meta, Story } from '@storybook/react';
import React from 'react';

import { BaseContainer } from '../../Layout/GlintsContainer/GlintsContainer';
import { Badge, BadgeProps } from './Badge';

(Badge as React.FunctionComponent<BadgeProps>).displayName = 'Badge';

const defaultText = 'Badge';

export default {
  title: '@next/Badge',
  component: Badge,
  decorators: [Story => <BaseContainer>{Story()}</BaseContainer>],
} as Meta;

const Template: Story<BadgeProps> = args => (
  <Badge {...args}>{defaultText}</Badge>
);

export const Interactive = Template.bind({});

Interactive.args = { status: null };
