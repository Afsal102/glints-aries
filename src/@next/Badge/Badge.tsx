import React from 'react';
import { Typography } from '../Typography';
import { Neutral } from '../utilities/colors';
import { BadgeStyle } from './BadgeStyle';

const badgeStatusVariant = [
  'neutral',
  'success',
  'informational',
  'warning',
  'critical',
  'promotion',
  'new',
] as const;

export type BadgeStatusVariant = typeof badgeStatusVariant[number];

export interface BadgeProps extends React.HTMLAttributes<HTMLDivElement> {
  children?: React.ReactNode;
  status?: BadgeStatusVariant;
}

export const Badge = ({ children, status, ...props }: BadgeProps) => {
  const content =
    typeof children === 'string' ? (
      <Typography
        variant="overline"
        color={status === 'new' ? Neutral.B100 : Neutral.B18}
        as={'span'}
      >
        {children}
      </Typography>
    ) : (
      children
    );

  return (
    <BadgeStyle status={status ? status : 'neutral'} {...props}>
      <div>{content}</div>
    </BadgeStyle>
  );
};
