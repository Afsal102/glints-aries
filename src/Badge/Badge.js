/* @flow */

import React from 'react';
import { Container } from '../Style/BadgeStyle';

const Badge = (props: Props) => {
  const {
    label,
    className,
    ...defaultProps
  } = props;

  return (
    <Container className={className} {...defaultProps}>
      {label}
    </Container>
  );
};

type Props = {
  label: string,
  className: string,
}

export default Badge;
