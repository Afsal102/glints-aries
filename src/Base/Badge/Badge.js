/* @flow */

import React, { Fragment } from 'react';
import { Container } from '../../Style/BadgeStyle';

const Badge = (props: Props) => {
  const {
    label,
    className,
    ...defaultProps
  } = props;

  return (
    <Fragment>
      {label
        && (
          <Container className={className} {...defaultProps}>
            {label}
          </Container>
        )
      }
    </Fragment>
  );
};

type Props = {
  label: string,
  className: string,
}

export default Badge;
