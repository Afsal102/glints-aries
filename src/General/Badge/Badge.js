/* @flow */

import React, { Fragment } from 'react';
import { BadgeContainer } from '../../Style/BadgeStyle';

const Badge = (props: Props) => {
  const {
    label,
    className,
    ...defaultProps
  } = props;

  return (
    <Fragment>
      {label
        && (
          <BadgeContainer className={className} {...defaultProps}>
            {label}
          </BadgeContainer>
        )
      }
    </Fragment>
  );
};

type Props = {
  label: string,
  className: string,
}

export default Badge;
