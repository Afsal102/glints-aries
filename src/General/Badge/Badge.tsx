import * as React from 'react';
import classNames from 'classnames';
import { isNil } from 'lodash';
import { BadgeContainer } from '../../Style/General/BadgeStyle';

const Badge: React.FunctionComponent<Props> = ({
  label, sup, className, ...defaultProps
}) => (
  <React.Fragment>
    {!isNil(label) && (
      <BadgeContainer className={classNames('aries-badge', className)} sup={sup} {...defaultProps}>
        <span className="badge-content">{label}</span>
      </BadgeContainer>
    )}
  </React.Fragment>
);

interface Props extends React.ComponentPropsWithoutRef<typeof BadgeContainer> {
  label: string | number;
  sup?: boolean;
}

export default Badge;
