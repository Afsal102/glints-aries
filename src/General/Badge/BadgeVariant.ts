export const BadgeVariant = {
  DEFAULT: 'default',
  DIMMED: 'dimmed',
};
