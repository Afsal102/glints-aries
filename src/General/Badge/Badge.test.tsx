import * as React from 'react';

import * as renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import Badge from './Badge';

const label = '6';

it(`<Badge> should render a label with the text "6"`, () => {
  const BadgeSnapshot = renderer.create(<Badge label={label} />).toJSON();
  expect(BadgeSnapshot).toMatchSnapshot();
});

it('should display the correct text when rendered', () => {
  const { queryByText } = render(<Badge label={label} />);
  const BadgeContent = queryByText(label);
  expect(BadgeContent).toBeVisible();
});

describe('when sup is:', () => {
  it('true, should have a top property', () => {
    const { getByRole } = render(<Badge label={label} sup />);
    const badgeContainer = getByRole('presentation');
    expect(badgeContainer).toHaveStyle('top: -.6em');
  });

  it('false, should not have a top property', () => {
    const { getByRole } = render(<Badge label={label} sup={false} />);
    const badgeContainer = getByRole('presentation');
    const { top } = getComputedStyle(badgeContainer);
    expect(top).toBeFalsy();
  });
});
