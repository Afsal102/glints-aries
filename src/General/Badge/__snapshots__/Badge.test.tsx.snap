// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge> should render a label with the text "6" 1`] = `
<div
  className="aries-badge BadgeStyle__BadgeContainer-sc-1rr32yz-0 haqjCy"
  role="presentation"
>
  <span
    className="badge-content"
  >
    6
  </span>
</div>
`;
